app-admin/etcd-wrapper/files/etcd-wrapper

```diff
@@ -1,76 +1,80 @@
-        --trust-keys-from-https
-    "
-fi
+ETCD_IMAGE_URL="${ETCD_IMAGE_URL:-quay.io/coreos/etcd}"
+ETCD_IMAGE="${ETCD_IMAGE:-${ETCD_IMAGE_URL}:${ETCD_IMAGE_TAG}}"
 
-if [[ ! -e ${ETCD_DATA_DIR} ]]; then
-    mkdir -p ${ETCD_DATA_DIR}
-    chown ${ETCD_IMG_USER} ${ETCD_DATA_DIR}
+if [[ "${ETCD_IMAGE%%/*}" == "quay.io" ]]; then
+	RKT_RUN_ARGS="${RKT_RUN_ARGS} --trust-keys-from-https"
 fi
 
-SYSTEMD_SYSTEM_DIR_SRC=${SYSTEMD_SYSTEM_DIR_SRC:-/run/systemd/system}
-if [[ -d ${SYSTEMD_SYSTEM_DIR_SRC} ]]; then
-    RKT_RUN_ARGS="${RKT_RUN_ARGS} \
-    --mount volume=systemd-dir,target=/run/systemd/system \
-    --volume systemd-dir,kind=host,source=${SYSTEMD_SYSTEM_DIR_SRC},readOnly=true \
-    "
+if [[ ! -e "${ETCD_DATA_DIR}" ]]; then
+	mkdir --parents "${ETCD_DATA_DIR}"
+	chown "${ETCD_USER}" "${ETCD_DATA_DIR}"
 fi
 
+ETCD_SSL_DIR="${ETCD_SSL_DIR:-/etc/ssl/certs}"
 
-if [[ -S ${NOTIFY_SOCKET} ]]; then
-    RKT_RUN_ARGS="${RKT_RUN_ARGS} \
-    --mount volume=notify,target=/run/systemd/notify \
-    --volume notify,kind=host,source=${NOTIFY_SOCKET} \
-    --set-env=NOTIFY_SOCKET=/run/systemd/notify \
-    "
+SYSTEMD_SYSTEM_DIR_SRC="${SYSTEMD_SYSTEM_DIR_SRC:-/run/systemd/system}"
+if [[ -d "${SYSTEMD_SYSTEM_DIR_SRC}" ]]; then
+	RKT_RUN_ARGS="${RKT_RUN_ARGS} \
+		--mount volume=systemd-dir,target=/run/systemd/system \
+		--volume systemd-dir,kind=host,source=${SYSTEMD_SYSTEM_DIR_SRC},readOnly=true \
+	"
 fi
 
-SSL_CERTS_SRC=${SSL_CERTS_SRC:-/etc/ssl/certs}
-ETC_HOSTS_SRC=${ETC_HOSTS_SRC:-/etc/hosts}
-ETCD_IMG_EXEC=${ETCD_IMG_EXEC:-/etcd}
+if [[ -S "${NOTIFY_SOCKET}" ]]; then
+	RKT_RUN_ARGS="${RKT_RUN_ARGS} \
+		--mount volume=notify,target=/run/systemd/notify \
+		--volume notify,kind=host,source=${NOTIFY_SOCKET} \
+		--set-env=NOTIFY_SOCKET=/run/systemd/notify \
+	"
+fi
 
-RKT=${RKT:-/usr/bin/rkt}
-RKT_STAGE1_ARG=${RKT_STAGE1_ARG:-"--stage1-from-dir=stage1-fly.aci"}
+RKT="${RKT:-/usr/bin/rkt}"
+RKT_STAGE1_ARG="${RKT_STAGE1_ARG:---stage1-from-dir=stage1-fly.aci}"
 set -x
 exec ${RKT} ${RKT_GLOBAL_ARGS} \
-    run ${RKT_RUN_ARGS} \
-    --volume data-dir,kind=host,source=${ETCD_DATA_DIR} \
-    \
-    --mount volume=etc-ssl-certs,target=/etc/ssl/certs \
-    --volume etc-ssl-certs,kind=host,source=${SSL_CERTS_SRC} \
-    \
-    --mount volume=etc-hosts,target=/etc/hosts  \
-    --volume etc-hosts,kind=host,source=${ETC_HOSTS_SRC} \
-    \
-    ${RKT_STAGE1_ARG} \
-    ${ETCD_IMG} ${ETCD_IMG_ARGS} \
-        --user=$(id -u ${ETCD_IMG_USER}) \
-        --exec=${ETCD_IMG_EXEC} -- \
-        "$@"
+	run ${RKT_RUN_ARGS} \
+	--volume data-dir,kind=host,source="${ETCD_DATA_DIR}",readOnly=false \
+	--volume etc-ssl-certs,kind=host,source="${ETCD_SSL_DIR}",readOnly=true \
+	--volume usr-share-certs,kind=host,source=/usr/share/ca-certificates,readOnly=true \
+	--volume etc-hosts,kind=host,source=/etc/hosts,readOnly=true \
+	--volume etc-resolv,kind=host,source=/etc/resolv.conf,readOnly=true \
+	--mount volume=data-dir,target=/var/lib/etcd \
```

Comment:
That's not right as well.
Imagine this:
- We overwrite ETCD_DATA_DIR from the calling unit to /mnt/etcd-node1
- rkt mounts /mnt/etcd-node1 into container as /var/lib/etcd
- etcd trues to load data in /mnt/etcd-node1 within the container.
Obviously it doesn't work because of --inherit-env

We either should mount it as ${ETCD_DATA_DIR} inside or (imho better) don't propagate this env variable to the container to avoid any permission errors and collisions.